plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraLatestConnectorSmokeTest.java

```diff
@@ -17,38 +17,72 @@
 import io.trino.testing.QueryRunner;
 import io.trino.testing.TestingConnectorBehavior;
+import org.testng.annotations.Test;
+
+import java.sql.Timestamp;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
 
 import static io.trino.plugin.cassandra.CassandraQueryRunner.createCassandraQueryRunner;
+import static io.trino.plugin.cassandra.CassandraTestingUtils.TABLE_DELETE_DATA;
+import static io.trino.plugin.cassandra.CassandraTestingUtils.createTestTables;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
 
 public class TestCassandraLatestConnectorSmokeTest
         extends BaseConnectorSmokeTest
 {
-    @Override
-    protected QueryRunner createQueryRunner()
-            throws Exception
-    {
-        CassandraServer server = closeAfterClass(new CassandraServer("3.11.9"));
-        return createCassandraQueryRunner(server, ImmutableMap.of(), REQUIRED_TPCH_TABLES);
-    }
+    private static final String KEYSPACE = "smoke_test";
+    private static final ZonedDateTime TIMESTAMP_VALUE = ZonedDateTime.of(1970, 1, 1, 3, 4, 5, 0, ZoneId.of("UTC"));
 
-    @Override
     protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
     {
         switch (connectorBehavior) {
             case SUPPORTS_CREATE_SCHEMA:
                 return false;
 
+            case SUPPORTS_CREATE_VIEW:
+                return false;
+
             case SUPPORTS_RENAME_TABLE:
+            case SUPPORTS_RENAME_TABLE_ACROSS_SCHEMAS:
                 return false;
 
-            case SUPPORTS_DELETE:
-                return true;
+            case SUPPORTS_ARRAY:
+                return false;
+
+            case SUPPORTS_COMMENT_ON_TABLE:
+            case SUPPORTS_COMMENT_ON_COLUMN:
+                return false;
 
-            case SUPPORTS_ROW_LEVEL_DELETE:
+            case SUPPORTS_TOPN_PUSHDOWN:
                 return false;
 
             default:
                 return super.hasBehavior(connectorBehavior);
         }
     }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        CassandraServer server = closeAfterClass(new CassandraServer("3.11.10"));
+        CassandraSession session = server.getSession();
+        createTestTables(session, KEYSPACE, Timestamp.from(TIMESTAMP_VALUE.toInstant()));
+        return createCassandraQueryRunner(server, ImmutableMap.of(), REQUIRED_TPCH_TABLES);
+    }
+
+    @Test
+    @Override
+    public void testDelete()
+    {
+        String keyspaceAndTable = format("%s.%s", KEYSPACE, TABLE_DELETE_DATA);
+        assertThat(query("SELECT * FROM " + keyspaceAndTable)).hasResultRowCount(15);
+
+        String wherePrimaryKey = " WHERE partition_one=3 AND partition_two=3 AND clust_one='clust_one_3'";
+        assertUpdate("DELETE FROM " + keyspaceAndTable + wherePrimaryKey);
+
+        assertThat(query("SELECT * FROM " + keyspaceAndTable)).hasResultRowCount(14);
```

Comment:
nit: Or you can run a `COUNT(*)` query instead like:

```java
assertQuery("SELECT COUNT(*) FROM " + keyspaceAndTable, "VALUES 14");
```

This will make the `hasResultRowCount` not needed (it is a good addition but with more number of ways to do the same thing the codebase becomes fragmented).

Reply:
Good point. Thanks for the heads up.